### PR TITLE
Some fixes for tests with tsan

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changes:
- Temporarily disable [the most flaky](https://play.clickhouse.com/play?user=play&url=https%3A%2F%2Fplay.clickhouse.com%2F#U0VMRUNUIGNoZWNrX25hbWUsIHRlc3RfbmFtZSBBUyBuYW1lLCBjb3VudCgpIEFTIGMsIGNvdW50RGlzdGluY3QocHVsbF9yZXF1ZXN0X251bWJlcikgYXMgY2QsIGFyZ01heChyZXBvcnRfdXJsLCBjaGVja19zdGFydF90aW1lKQpGUk9NIGNoZWNrcwpXSEVSRQogICAgY2hlY2tfbmFtZSBMSUtFICclU3RhdGUldGhyZWFkJScKICAgIEFORCAodGVzdF9zdGF0dXMgTElLRSAnRiUnIE9SIHRlc3Rfc3RhdHVzIExJS0UgJ0UlJykKICAgIEFORCBjaGVja19zdGFydF90aW1lID49IHRvZGF5KCkgLSA3CkdST1VQIEJZIGNoZWNrX25hbWUsIG5hbWUKSEFWSU5HIG1heChjaGVja19zdGFydF90aW1lKSA+PSBub3coKSAtIElOVEVSVkFMIDI0MCBIT1VSCk9SREVSIEJZIGMgREVTQw==) functional tests
- Temporarily disable ThreadFuzzer with ThreadSanitizer (#38019)
- Add C11 threading primitives to the list of harmful functions, rename confusing functions in librdkafka (#36547)

**It does not fix random timeouts in checks with tsan, the reason of these failures is still unknown.**

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Thread Fuzzer is especially good in a combination with Thread Sanitizer.
If we will disable it, it is a step backward, and there is a chance that we will forget about it.

Maybe simply build the latest LLVM, Clang, and compiler-rt from sources (on Docker container create)?
They almost have this fix... https://github.com/google/sanitizers/issues/1540

**Comment by alexey-milovidov** [comment]
And we can have both clangs in the CI: clang-14 for most of the cases and clang-15-master for TSan.

File: tests/queries/0_stateless/01183_custom_separated_format_http.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env bash
+# Tags: no-tsan
+# FIXME should work with tsan
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/01183_custom_separated_format_http.sh:3]: If someone from the outside will read this comment, they can get the wrong impression that we have some code that does not pass TSan tests. But ClickHouse is 100% clean. Let's add a more descriptive comment.

**Comment by tavplubix** [comment]
> Thread Fuzzer is especially good in a combination with Thread Sanitizer.
> If we will disable it, it is a step backward

It's not obvious, because builds with tsan are even slower than debug builds sometimes and ThreadFuzzer makes them much more slower. We had bugs that were reproducing only is stress tests with ubsan, because it's the fastest sanitizer. Maybe tsan will be less effective without fuzzer, but at least it will run more tests per one check run.

> Maybe simply build the latest LLVM, Clang, and compiler-rt from sources (on Docker container create)?

Does not sound simple =) Also using llvm from master may introduce more obscure issues.
And maybe they will backport the fix to llvm-14.

**Comment by alesapin** [comment]
Why we had to do something with `rdkafka`?

**Comment by tavplubix** [comment]
> Why we had to do something with `rdkafka`?

See https://github.com/ClickHouse/ClickHouse/issues/36547. We need to make sure that we don't use C11 threading primitives

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Need to figure out, what's with 'test_hive_query' and LGTM.

**Comment by alesapin** [comment]
`test_hive_query` is flaky https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZEYXkoY2hlY2tfc3RhcnRfdGltZSkgYXMgZCwKY291bnQoKSwgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlciksIGdyb3VwVW5pcUFycmF5KGNoZWNrX25hbWUpLCBhbnkocmVwb3J0X3VybCkKZnJvbSBjaGVja3Mgd2hlcmUgJzIwMjItMDQtMDEnIDw9IGNoZWNrX3N0YXJ0X3RpbWUgYW5kIHRlc3RfbmFtZSBsaWtlICcldGVzdF9oaXZlX3F1ZXJ5JScgYW5kIHRlc3Rfc3RhdHVzIG5vdCBpbiAoJ09LJywgJ1NLSVBQRUQnKSBncm91cCBieSBkIG9yZGVyIGJ5IGQgZGVzYw==

But this fix is too important. Let's merge it.

**Pull request merged**